src/ocl/convolutionocl.cpp

```diff
@@ -826,55 +827,99 @@ void ConvolutionDescriptor::GetSolutionsFallback(Handle& handle,
     ctx.SetStream(&handle);
     ctx.DetectRocm();
 
-    const auto wti2time = [](const float& wti) {
-        assert(wti != 0.0f);
-        if(wti <= 0.0f) // Return negative values as is, avoid DIV/0.
-            return wti;
-        return 10.0f / wti; // Assume WTI == 1.0 (100%) is 10 ms.
-    };
-
-    for(const auto& solver_id : solver::GetSolversByPrimitive(solver::Primitive::Convolution))
+    if(MIOPEN_ENABLE_AI_HEUR &&
+       ConvHeur::IsHeurApplicable(handle.GetDeviceName(), problem.conv_problem))
     {
-        // solver_id is always valid here, because taken from registry.
-        // Validity check is not required.
-        const auto algo = solver_id.GetAlgo();
-        if(IsAlgorithmDisabled(algo)) // Algos can be disabled globally.
-            continue;
-        const auto& s = solver_id.GetSolver();
-        if(s.IsEmpty())
-            continue;
-        if(!s.IsDynamic()) // Let's allow non-dynamic later, if necessary.
-            continue;
-        if(!s.IsApplicable(ctx))
-            continue;
+        int idx = 1; // Each solution to have a successively more negative values keeping
+        // sorting logic intact in frameworks
+        bool is_cached = false;
+        auto solvers = ConvHeur{}.Estimate(handle.GetDeviceName(), problem.conv_problem, is_cached);
+        for(const auto kinder : solvers)
+        {
+            const auto solver_id = solver::Id{kinder};
+            const auto sol       = solver_id.GetSolver();
+            if(!sol.IsDynamic())
+                continue; // branch should never be taken
+            if(!sol.IsApplicable(ctx))
+                continue;
+            const auto algo = solver_id.GetAlgo();
+            if(IsAlgorithmDisabled(algo))
+                continue;
+            interim.emplace_back(
+                static_cast<float>(-1 * idx), sol.GetWorkspaceSize(ctx), solver_id.Value(), algo);
```

Comment:
Let's have a function similar to `wti2time` that is intended to transform AI index to estimated time.

Reply:
added the function in my refactor for AI timing.